x-pack/plugins/canvas/public/state/actions/elements.js

```diff
@@ -213,12 +214,48 @@ export const duplicateElement = createThunk(
   }
 );
 
+export const rawDuplicateElement = createThunk(
+  'rawDuplicateElement',
+  ({ dispatch, type }, element, pageId, root) => {
+    const newElement = cloneDeep(element);
+    // move the root element so users can see that it was added
+    newElement.position.top = newElement.position.top + 10;
+    newElement.position.left = newElement.position.left + 10;
+    const _rawDuplicateElement = createAction(type);
+    dispatch(_rawDuplicateElement({ pageId, element: newElement }));
+
+    // refresh all elements if there's a filter, otherwise just render the new element
+    if (element.filter) {
+      dispatch(fetchAllRenderables());
+    } else {
+      dispatch(fetchRenderable(newElement));
+    }
+
+    // select the new element
+    if (root) {
+      window.setTimeout(() => dispatch(selectElement(newElement.id)));
```

Comment:
Why does this need to be wrapped in `setTimeout`? What happens if you dispatch immediately?

Reply:
I want to circle back to this one, the element in `aeroelastic` is not present yet if done synchronously, so it can't select (pasting will create a new group, naturally with a new `id`). So the purpose is to switch selection to the newly pasted thing, even if it's a group. I think it's due to the fact that cloning is done outside `aero` but selection is in good part, inside (eg. to activate the resize frame). I'm sometimes thinking about a different synchronization between `aero` and the Redux actions / thunks, it got fairly convoluted. Something I'm planning to touch on on our Jan 10 discussion on the layout engine.